In [5]:
from reaper_project import ReaperProject, NodeType
from tokenizer import tokenize_node

In [3]:
from pathlib import Path

In [4]:
files = [
    'D:/41-50_MUSIC/41_LAB\Вдохновения/Electroshit/Dark Ambient/Dark Ambient.rpp',
    'D:/41-50_MUSIC/41_LAB/Вдохновения/Tbilisi Acoustic G/221012 01 Tbilisi Sunrise/221012 01 Zoom Rec Again.rpp',
    'D:/41-50_MUSIC/41_LAB/230416 01 mary composes/230416 01 mary composes.rpp'
]

In [26]:
%%timeit -r 50 -n 10
# p = Path('D:/41-50_MUSIC/41_LAB/')

# for fp in p.glob("**/*.rpp"):
# for fp in files:
    # print("="*20 + " " + str(fp))
    # rp = ReaperProject(fp)
    # rp.head.print_tree()
    # fetch_res = rp.head.fetch(NodeType.VST, [])
    # for res in fetch_res:
        # print(res.parameters_first_line[0])
        # res.print_tree()
    # rp = tokenize_node(open(fp, 'r').readlines(), supress_output=True)
    # vsts = []
    # rp[0].fetch_nodes(name='VST', collection=vsts)
    

rp = ReaperProject(files[2])
# rp = tokenize_node(open(files[2], 'r').readlines(), supress_output=True)

30.2 ms ± 3.63 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)


// Reaper Project, with glob, with VST Search, with Prints // 2.58 s ± 75.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
// Reaper Project, with glob, with VST Search, w/o Prints // 2.41 s ± 47.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
// Reaper Project, with glob, w/o VST Search, w/o Prints // 2.37 s ± 79.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)



// Reaper Project, w/o glob, 3 predefined test projects, w/o VST Search, w/o Prints // 335 ms ± 7.35 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)
// Reaper Project, w/o glob, 3 predefined test projects, w/ VST Search, w/o Prints // 346 ms ± 15.3 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)

// Tokenizer, w/o glob, 3 predefined test projects, w/o VST Search, w/o Prints // 74.1 ms ± 3.45 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)
// Tokenizer, w/o glob, 3 predefined test projects, w/ VST Search, w/o Prints // 77.3 ms ± 3.29 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)


// Reaper Project, Test Project 1, w/o VST Search, w/o Prints // 48.8 ms ± 2.65 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)
// Tokenizer, Test Project 1, w/o VST Search, w/o Prints // 5.34 ms ± 657 µs per loop (mean ± std. dev. of 50 runs, 10 loops each)

// Reaper Project, Test Project 2, w/o VST Search, w/o Prints // 260 ms ± 10 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)
// Tokenizer, Test Project 2, w/o VST Search, w/o Prints // 67.7 ms ± 3.04 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)

// Reaper Project, Test Project 3, w/o VST Search, w/o Prints // 30.2 ms ± 3.63 ms per loop (mean ± std. dev. of 50 runs, 10 loops each)s
// Tokenizer, Test Project 3, w/o VST Search, w/o Prints // 3.25 ms ± 379 µs per loop (mean ± std. dev. of 50 runs, 10 loops each)